### ISSUE TITLE
Add ARM-Devices support

### ISSUE THREAD

**Issue by dockerpirate** [roundcube/roundcubemail-docker#122]
Please add arm-devices support, it works easy with docker buildx

https://docs.docker.com/docker-for-mac/multi-arch/

**Comment by amiga23** [comment]
Can not build it on aarch64 (apache/Dockerfile), do get the following error message and have no glue what that should mean:
```
+ docker-php-ext-configure gd --with-freetype-dir=/usr/include/ --with-png-dir=/usr --with-jpeg-dir=/usr
fetch https://dl-cdn.alpinelinux.org/alpine/edge/main/aarch64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/edge/community/aarch64/APKINDEX.tar.gz
ERROR: unsatisfiable constraints:
  pkg-config (missing):
    required by: .phpize-deps-configure-20210203.181158[pkg-config]
error building image: error building stage: failed to execute command: waiting for process to exit: exit status 2
```

**Comment by dockerpirate** [comment]
Alpine is very specific, it could be, that some packages are missing or have different names, or other odd issues

**Comment by jdavidbakr** [comment]
Any chance this is going to get an arm64 docker image soon?  I'm wanting to deploy this into my AWS ECS environment but it's failing because it's running in the arm64 architecture.

**Comment by thomascube** [comment]
@jdavidbakr I'm afraid not. We'd welcome pull requests for this but we're not actively develop an arm branch at this time.

**Comment by williamdes** [comment]
I can work on this, are contributions welcome to make the roundcube image work for as much archs as possible ?

I would of course use docker buildx, see examples in: https://github.com/sudo-bot/nut-upsd

**Comment by thomascube** [comment]
@williamdes Yes, contributions are very welcome. So far we're building the images directly on Docker Hub via Webhooks. We should check whether that's still possible for other archs, too.

**Comment by williamdes** [comment]
> @williamdes Yes, contributions are very welcome. So far we're building the images directly on Docker Hub via Webhooks. We should check whether that's still possible for other archs, too.

Well I think this will not be possible but the easy alternative is to make the webhooks trigger a workflow in this repository that will launch the build
And even easier if you push a tag for each version on this repository the workflow would just have to wait for events on tags.

Let me know what you prefer

Docker buildx does not work with Docker Hub as far as I know
But GitHub actions does the job perfectly to build images and push them !

**Comment by thomascube** [comment]
> Docker buildx does not work with Docker Hub as far as I know
> But GitHub actions does the job perfectly to build images and push them !

That's fair enough. I think we can go with this. Let's first get the Dockerfiles right and then see how we can best automate the builds. Thanks a lot for your efforts!

**Comment by williamdes** [comment]
> Hi there,
As one of the users of Docker Hub’s Autobuild service, we wanted to reach out and let you know that from June 18th, 2021 users on the free plan will no longer have access to the Autobuild feature. This means that any of your currently configured Autobuild will fail to run and you will need to upgrade to a Pro or Team account to restore functionality. You can read more about the changes on our blog.

I received this from Docker hub
I suppose this will be a good occasion to fix two problems in one time

**Comment by thomascube** [comment]
@williamdes yes, i saw that announcement too. We will, however, apply for the open source program but in parallel I’ll also read into Github’sDocker services.

**Comment by williamdes** [comment]
> @williamdes yes, i saw that announcement too. We will, however, apply for the open source program but in parallel I’ll also read into Github’sDocker services.

I will be contributing soon to fix this arm problem :)

I just deployed my [Docker Mailserver](https://github.com/docker-mailserver/docker-mailserver#readme) replica on my RaspberryPi

**Comment by AlexanderZhirov** [comment]
> I just deployed my [Docker Mailserver](https://github.com/docker-mailserver/docker-mailserver#readme) replica on my RaspberryPi

Me too. Roundcube is very necessary

**Comment by williamdes** [comment]
Starting to work on it

will update this summary:
- [x] make it build on GitHub actions: https://github.com/williamdes/roundcubemail-docker/actions/runs/1139193225
- [x] drop TravisCI and Docker Hub hooks
- [x] drop useless docker-compose files
- [ ] rebuild a plausible test
- [ ] make multi arch builds

**Comment by williamdes** [comment]
Hi @J0WI 
I have re-worked the process in https://github.com/roundcube/roundcubemail-docker/pull/141

Seeing the monstrous time it takes to build multi arch: https://github.com/williamdes/roundcubemail-docker/actions/runs/1144445590
I would consider making Roundcube apply to the Docker official repository program that would allow cost-less multi arch builds in the end

**Comment by Etruscian** [comment]
What is the current state of the arm image? I don't have a lot of docker image build knowledge, but if I can help, please do tell! :)

**Comment by seb-daehne** [comment]
Hey,

it looks like GitHub Actions are now used on mainline - is the plan to add the arm build or do you intend to stay on dockerhub builds?

I believe there you can also build for arm.

**Comment by williamdes** [comment]
Hi @J0WI 
Would you mind helping roundcube have a docker official image ?

**Comment by azoller1** [comment]
For anyone wanting to build for arm/arm64, it is very easy. Just clone the repo and choose a version/directory and build form the dockerfile. I built directly on my pi 4 with buildx.

Install the buildx alias first
`docker buildx install`

Then just run the docker build command (fpm one worked for me)
`docker build -f Dockerfile . -t user/roundcube:version`

**Comment by ZsBT** [comment]
Github does support buildx. I already use it to make multi-arch images with this action: https://github.com/ZsBT/haproxy-quic/blob/main/.github/workflows/docker-image.yml
I've built an image on my arm server using the official Dockerfile and works like a charm.

I'd be happy to help if you accept PRs.

**Comment by williamdes** [comment]
> Github does support buildx. I already use it to make multi-arch images with this action: https://github.com/ZsBT/haproxy-quic/blob/main/.github/workflows/docker-image.yml I've built an image on my arm server using the official Roundcube Dockerfile and works like a charm.
> 
> I'd be happy to help if you accept PRs.

I think you can open a PR using buildx
Until this image eventually gets to official images

**Comment by ZsBT** [comment]
PR #200 sent, workflow awaiting approval

**Comment by williamdes** [comment]
Implemented in #201
And pushed to the hub, so /closed?

**Comment by ZsBT** [comment]
I've just tested on both 64-bit and 32-bit arm systems, the containers start properly.

**Comment by williamdes** [comment]
Hi @dockerpirate 
Can you close this issue ?
